go-controller/pkg/clustermanager/secondary_network_cluster_manager.go

```diff
@@ -114,6 +114,8 @@ func (sncm *secondaryNetworkClusterManager) isTopologyManaged(nInfo util.NetInfo
 	switch nInfo.TopologyType() {
 	case ovntypes.Layer3Topology:
 		return true
+	case ovntypes.Layer2Topology:
+		return config.OVNKubernetesFeature.EnableInterconnect
```

Comment:
why is this decided based on if interconnect is enabled for not? If not IC, then allocation is handled somewhere else? where is that done?

Reply:
That's done by ovnkube-master, in `(bnc *BaseNetworkController) allocatePodAnnotationForSecondaryNetwork`. There we have a reciprocal check.

I will add comments here though.